### PR TITLE
Add bundle document

### PR DESCRIPTION
**Pull request by makyo**
Add a document describing the ways in which the charmstore treats bundles.  Preview [here](https://github.com/makyo/charmstore/blob/bundledoc/docs/bundles.md)

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/837/
Test PASSed.

**Comment by hatched** [comment]
This looks really good but I'd like to see an example bundle which included all of the supported features

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/838/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/839/
Test PASSed.

**Comment by hatched** [comment]
Great thanks for that addition! :+1:

File: docs/bundles.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+Version 3 bundles are currently existing bundles that specify a deployment as a
+list of services and, optionally, relations.  The charmstore will not support
+the idea of a "basket" or multiple bundles within one file.  However, existing
+baskets will still be imported, and split up into their component bundles.
+
+## Version 4 bundles
+
+Version 4 bundles are identical to version 3 bundles except for a few key
+differences: the `branch` attribute of the service spec is no longer supported,
+they must contain a machine specification, and their deployment directives are
```
Comment (frankban) [inline docs/bundles.md]: The machines section is not mandatory IIRC.

File: docs/bundles.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+different from version 3 bundles.  A bundle without a machine specification is
```
Comment (frankban) [inline docs/bundles.md]: This is an internal impl detail of the juju deployerm not sure if it belongs here. Maybe in its own paragraph about the deployer? I'd like not to pass the message that bundles v4 strictly require a machine spec.

**Comment by frankban** [comment]
:+1: with just a minor note.
Thank you!

File: docs/bundles.md (hunk 2 above)
Comment (makyo) [inline docs/bundles.md]: Good point; I've moved this to a separate paragraph about deploying the bundles with the deployer and GUI.

**Comment by makyo** [comment]
Thanks all. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/841/
Test PASSed.

**Pull request merged**